sbor/src/payload_validation/traits.rs

```diff
@@ -1,13 +1,20 @@
 use crate::traversal::*;
 use crate::*;
 
-pub trait ValidatableCustomTypeExtension<T>: CustomTypeExtension {
+pub trait ValidatableCustomExtension<T>: CustomExtension {
     // Note that the current SBOR extension only supports terminal custom type,
     // i.e., no custom value can be container.
 
-    fn apply_custom_type_validation<'de>(
-        custom_type_validation: &Self::CustomTypeValidation,
-        value: &TerminalValueRef<'de, Self::CustomTraversal>,
-        context: &mut T,
-    ) -> Result<(), ValidationError>;
+    fn apply_validation_for_custom_value<'de>(
```

Comment:
Glad we moved away from `TypeValidation`-driven validation to `Value`-driven validation.

Reply:
Haha yes, it's kinda a little more intuitive, I guess?

The value-driven validation slightly hides the fact that anything passes `TypeValidation::None` and the fact that a mismatch between TypeValidation and TypeKind is a SchemaValidation

BUT I think it's just generally clearer/more intuitive, I agree. And actually in this Manifest case, it's required :)